sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -138,8 +138,12 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
 
       withProjection :: Nil
 
-    case WriteToDataSourceV2(writer, query) =>
-      WriteToDataSourceV2Exec(writer, planLater(query)) :: Nil
+    case WriteToDataSourceV2(relationOpt, writer, query) =>
+      val refreshCacheFunc: () => Unit = relationOpt match {
+        case Some(r) => refreshCache(r)
```

Comment:
I'm not sure if refresh cache is the best choice in the context of streaming - perhaps we should invalidate it?

Reply:
If you have a concern, why don't we have a config for that? We can have a controllability to invalidate or to refresh.

Reply:
I think refreshing the cache per second (assuming the streaming trigger is one second) doesn't make sense. Invalidating seems more reasonable.

Reply:
Hmm, I cannot think about a use-case we want to refresh the cache of a table written by streaming.

Reply:
Thanks all! Let me switch to invalidating cache. IMO a config isn't necessary since ppl may never want this behavior (it could get very expensive).